spec/controllers/votes_controller_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe VotesController, type: :controller do
+  let(:quote) { create(:quote) }
+  let(:user) { create(:user) }
+  let(:vote) { create(:vote, user: user, quote: quote) }
+
+  describe "#create" do
+    subject { post :create, params }
+    context 'without a signed in user' do
+      let(:params) { { quote_id: quote } }
+      it { is_expected.to redirect_to new_user_session_path }
+    end
+    context "with a signed in user" do
+      before { login_with user }
+      context "with valid parameters" do
+        let(:params) { { quote_id: quote.id, vote: attributes_for(:vote) } }
+        it { is_expected.to redirect_to quotes_path }
+
+        it "creates a vote associated with the user in the database" do
+          expect { subject }.to change { user.votes(true).count }.from(0).to(1)
```

Comment:
While this is good it could be fragile if something in the code implementation causes a vote to be initialized before the subject is run. An alternative is 

`expect { subject }.to change { user.votes(true).count }.by 1`